### DISCUSSION TITLE
PDF preview failure at WSL

### DISCUSSION THREAD

**Discussion by alanlivio** [James-Yu/LaTeX-Workshop#3833] [Q&A] [answered]
Hi everyone. I created issue #3826 about the PDF preview failure at WSL, and the maintainer indicated a firewall issue.
By navigating the previous issues, I found these related:  #3664, #3008, #3242, #2799, #3668
I want to ask if anyone has found a solution. 
I particularly ask the creators of the issues @nidrissi, @tamuratak, @amyspark, @blackandcold, @zhxchd.
Since now, many thanks.

**Comment by nidrissi** [comment]
It's not a firewall issue (the comment I made about a firewall was misinterpreted by the developers). It's a bug in WSL. https://github.com/microsoft/WSL/issues/9508 It was fixed in WSL 1.1.2, but the bug apparently came back in 1.2.0.

**Reply by nidrissi** [reply]
See also https://github.com/microsoft/WSL/issues/9746

**Reply by alanlivio** [reply]
Many thanks for your answer @nidrissi. 

I installed the WSL 1.1.2 from [here](https://github.com/microsoft/WSL/releases/tag/1.1.2), did `rm -rf .vscode-server/` at WSL, and reinstalled LaTeX-workshop, but the problem persists =(
Do you have another suggestion?

```ps1
> wsl --version
WSL version: 1.1.2.0
Kernel version: 5.15.83.1
WSLg version: 1.0.49
MSRDC version: 1.2.3770
Direct3D version: 1.608.2-61064218
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.22624.1470
> cat C:\Users\alan\.wslconfig
[wsl2]
guiApplications=true
> wsl --list --verbose
  NAME                   STATE           VERSION
* Ubuntu                 Running         2
```

**Reply by nidrissi** [reply]
I think I misremembered and it was fixed in 1.1.3. They had a first shot in 1.1.2 and it fixed other issues but not LW.

**Reply by nidrissi** [reply]
I opened an issue there: https://github.com/microsoft/WSL/issues/9947

**Reply by alanlivio** [reply]
Many thanks; I will follow the issue.

**Reply by nidrissi** [reply, marked as answer]
It's fixed in WSL 1.2.1.0.

**Reply by alanlivio** [reply]
Thank you so much, @nidrissi. The WSL 1.2.1 worked. 
If any wants to try, I get it from [here ](https://github.com/microsoft/WSL/releases/tag/1.2.1).
